### PR TITLE
Henryh/segmentation metrics

### PR DESCRIPTION
**Pull request by Hgherzog**
Goal: Be able to compare segmentation metrics for LP with metrics we get in rslearn

Example on Pastis: https://wandb.ai/eai-ai2/2025_01_30_validate_seg_metrics?nw=nwuserhenryhzog

### Summary
Adds additional segmentation metrics beyond mIoU. All eval functions (`run_knn`, `train_and_eval_probe`, `run_finetune_eval`) now return a unified `EvalResult` dataclass containing:

- **miou** (primary metric, used for model selection)
- **overall_acc** - total correct / total pixels
- **macro_acc** - mean per-class recall  
- **macro_f1** - mean per-class F1

### Changes
- New `EvalResult` dataclass in `metrics.py` with factory methods for classification and segmentation
- New `segmentation_metrics()` function computes all metrics from a single confusion matrix pass
- Updated `knn.py`, `linear_probe.py`, `finetune.py` to return `EvalResult`
- Updated `evaluator_callback.py` to log all metrics to wandb (e.g., `eval/pastis/miou`, `eval/pastis/macro_f1`)
- Backward compat: still logs primary metric at `eval/{task_name}` for existing dashboards

### Breaking Change
Eval functions now return `EvalResult` objects instead of `float`. Callers should use:
result = run_finetune_eval(...)
score = result["val_score"].primary  # or .metrics["miou"]

### CODE REVIEW

File: olmoearth_pretrain/evals/finetune.py [hunk 1]
```diff
@@ -396,23 +405,18 @@ def run_finetune_eval(
...
+            test_result = _eval_cls(ft, test_loader, device, task_config.is_multilabel)
+        else:
+            test_result = _eval_seg(
+                ft, test_loader, device, task_config.num_classes, patch_size
+            )
 
     return {
-        "val_score": val_score,
-        "test_score": test_score,
+        "val_score": best_val_result,
```
Comment (favyen2) [inline olmoearth_pretrain/evals/finetune.py]: Consider renaming the keys in this dict too (or make it structured class).

File: olmoearth_pretrain/evals/finetune.py [hunk 2]
```diff
@@ -125,15 +125,16 @@ def _eval_cls(
     labels = torch.cat(labels_all, 0)
     if is_multilabel:
         preds = torch.sigmoid(logits).gt(0.5).int()
-        return f1_score(
+        score = f1_score(
```
Comment (favyen2) [inline olmoearth_pretrain/evals/finetune.py]: It may make sense to update this to compute simple accuracy for all tasks and report F1 in addition for multi-label tasks. Currently it is a bit confusing that F1 score is called accuracy since accuracy usually refers to a different metric.

**Review by favyen2 — APPROVED** [review]
